src/core/server/saved_objects/service/lib/repository.ts

```diff
@@ -151,6 +151,7 @@ export interface SavedObjectsDeleteByNamespaceOptions extends SavedObjectsBaseOp
 }
 
 export const DEFAULT_REFRESH_SETTING = 'wait_for';
+export const DEFAULT_RETRY_COUNT = 3;
```

Comment:
I'm pretty sure I've raised this before but doing so again now:
Should we consider centralizing the Saved Objects constants somewhere so that they're easier to find?
We may not need to for all of them but for things like retry counts (especially when we set defaults), they'll be easier to find down the line.

Reply:
> a comment explaining the choice of 3 as a default number of retries would be good

That's actually a very good remark :) And I'm not even sure `3` is a good default tbh. Maybe `1` is just sufficient btw. @rudolf @TinaHeiligers WDYT?

> Should we consider centralizing the Saved Objects constants somewhere so that they're easier to find?

These constant are (at least atm) only used internally by the `SOR` and 'extracted' SOR API such as `src/core/server/saved_objects/service/lib/update_objects_spaces.ts`. But that's a good remark, I can create a `constant` file for them.

Reply:
When you don't specify a version you are opting into "last write wins". And retry_on_conflict is basically a way to force your write to succeed when there was a write between the time ES retrieved the document and then tried to write it back as part of the internal ES operation. If a plugin anyway wants last write wins, then doing this retry inside ES is much faster than going back to Kibana and therefore more likely to succeed.

3 seems like a reasonable default to me, when someone doesn't specify a version they're not going to build in error handling code to deal with a conflict. So realistically, I suspect a conflict that isn't retried leads to some kind of data loss or a cryptic error which the user can't act upon.

But in the extreme, if we retry 10k times, then it means there has been 9999 successful writes before our write succeeds, so when ours succeed we potentially loose data from those 9999 writes.

Using "last write wins" is generally acceptable if your data is relatively up to date, but the older it gets the more dangerous it could be.

After removing the concurrency control of the preflight check we've dramatically reduced the chance of a conflict without a version. So my guess would be 3 is enough to reduce the chances to practically 0, if a plugin does a huge amount of concurrent updates with last write wins, they should probably just ignore conflict errors because there's anyway another write that's going to overwrite the previous one.